File: string/1668_maximum_repeating_substring/1668_maximum_repeating_substring.py

class Solution:
    def maxRepeating(self, sequence: str, word: str) -> int:
        ans = 1
        while sequence.find(word * ans) >= 0:
            ans += 1
        return ans - 1


sequence = "ababc"
word = "ab"
res = Solution().maxRepeating(sequence, word)
print(res)

File: string/1668_maximum_repeating_substring/1668_maximum_repeating_substring.go
package main

import (
	"fmt"
	"strings"
)

func maxRepeating(sequence string, word string) int {
	var w = word
	var count = 0
	for strings.Contains(sequence, word) {
		count++
		word = strings.Repeat(w, count)
	}

	return (len(word) - len(w)) / len(w)
}

func main() {
	sequence := "ababc"
	word := "ab"
	res := maxRepeating(sequence, word)
	fmt.Println(res)
}
